### PR TITLE
Upgrade Dotty to 3.0.0-RC2

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: apps/resources/scala3-compiler.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "central"
   ],
   "dependencies": [
-    "org.scala-lang:scala3-compiler_3.0.0-RC1:3.0.0-RC1"
+    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC1"
```
Comment (alexarchambault) [inline apps/resources/scala3-compiler.json:7]: The version needs to be updated at the end too:
```suggestion
    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC2"
```

File: apps/resources/scala3-decompiler.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "central"
   ],
   "dependencies": [
-    "org.scala-lang:scala3-compiler_3.0.0-RC1:3.0.0-RC1"
+    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC1"
```
Comment (alexarchambault) [inline apps/resources/scala3-decompiler.json:7]: Same as above:
```suggestion
    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC2"
```

File: apps/resources/scala3-doc.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "central"
   ],
   "dependencies": [
-    "org.scala-lang:scala3-doc_3.0.0-RC1:3.0.0-RC1"
+    "org.scala-lang:scala3-doc_3.0.0-RC2:3.0.0-RC1"
```
Comment (alexarchambault) [inline apps/resources/scala3-doc.json:7]: Same as above:
```suggestion
    "org.scala-lang:scala3-doc_3.0.0-RC2:3.0.0-RC2"
```

File: apps/resources/scala3-repl.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "central"
   ],
   "dependencies": [
-    "org.scala-lang:scala3-compiler_3.0.0-RC1:3.0.0-RC1"
+    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC1"
```
Comment (alexarchambault) [inline apps/resources/scala3-repl.json:7]: Last one:
```suggestion
    "org.scala-lang:scala3-compiler_3.0.0-RC2:3.0.0-RC2"
```

**Comment by anatoliykmetyuk** [comment]
Oops, missed those trailing versions 😅

**Review by alexarchambault — APPROVED** [review]
Thanks @anatoliykmetyuk!

**Pull request merged**